File: Contracts/Model/Product/Response/DiscountResponse.cs
namespace Contracts.Model.Product.Response;

public class DiscountResponse
{
	public Guid Id { get; set; }
	public double Value { get; set; }
	public DateTimeOffset Start { get; set; }
	public DateTimeOffset End { get; set; }
	public ICollection<HotestProductsResponse>? ProductsDiscounts { get; set; }
}

File: ClientApp/src/app/core/interfaces/Product/Response/discountResponse.ts
import {OptionPreviewResponse} from "@core/interfaces/Product/Response/optionPreviewResponse";

export interface DiscountResponse {
  id: string;
  value: number;
  productsDiscounts: HotestProductsResponse[];
  start: Date;
  end: Date;
}

export interface HotestProductsResponse {
  option: OptionPreviewResponse;
}
